Ex6/Ex6.java

```diff
@@ -0,0 +1,118 @@
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+
+public class Ex6 {
+	private int availableSpot;
+	private int carToMove;
+	private int[] initial;
+	private int[] desired;
+	private ArrayList<Move> moves;
+	
+	public Ex6(int[] initial, int[] desired) {
+		this.initial = Arrays.copyOf(initial, initial.length);
+		this.desired = Arrays.copyOf(desired, desired.length);
+		moves = moveCars();
```

Comment:
In this line you assign to moves the return value of your function. But in your method doIfZero you try to access moves which would throw a NullPointerException. In your method moveCars you create a new ArrayList moves. Maybe it would solve the problem if you use this.move there as well..